### PR TITLE
fix(volume/attach): fix the problems of volume attachment and detachment

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The OpenStack API does not support EPS authentication, therefore, replacae these API with HuaweiCloud API (v2 to v1).

**Which issue this PR fixes**:
```
NONE
```

**Special notes for your reviewer**:

**Release note**:

```release-note
1. fix the problems of volume attachment and detachment for EPS authentication.
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccComputeV2VolumeAttach_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccComputeV2VolumeAttach_basic -timeout 360m -parallel 4
=== RUN   TestAccComputeV2VolumeAttach_basic
=== PAUSE TestAccComputeV2VolumeAttach_basic
=== CONT  TestAccComputeV2VolumeAttach_basic
--- PASS: TestAccComputeV2VolumeAttach_basic (248.59s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       248.690s
```

### CODE REVIEW

**Comment by Jason-Zhang9309** [comment]
Can we update the resource with context and diag by the way?

File: huaweicloud/resource_huaweicloud_compute_volume_attach.go [hunk 1]
```diff
@@ -16,11 +17,11 @@ import (
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
 )
 
-func ResourceComputeVolumeAttachV2() *schema.Resource {
+func ResourceComputeVolumeAttach() *schema.Resource {
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_compute_volume_attach.go:20]: If we changed the function name, joined providers should also be modified accordingly.

File: huaweicloud/resource_huaweicloud_compute_volume_attach.go [hunk 2]
```diff
@@ -6,8 +6,9 @@ import (
 	"time"
 
 	"github.com/chnsz/golangsdk"
-	"github.com/chnsz/golangsdk/openstack/compute/v2/extensions/volumeattach"
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_compute_volume_attach.go:9]: can we totally replace this package?

File: huaweicloud/resource_huaweicloud_compute_volume_attach.go [hunk 3]
```diff
@@ -182,50 +180,17 @@ func resourceComputeVolumeAttachV2Delete(d *schema.ResourceData, meta interface{
...
-	computeClient *golangsdk.ServiceClient, instanceId, attachmentId string) resource.StateRefreshFunc {
+func attachmentJobRefreshFunc(c *golangsdk.ServiceClient, jobId string) resource.StateRefreshFunc {
 	return func() (interface{}, string, error) {
-		logp.Printf("[DEBUG] Attempting to detach HuaweiCloud volume %s from instance %s",
-			attachmentId, instanceId)
-
-		va, err := volumeattach.Get(computeClient, instanceId, attachmentId).Extract()
+		resp, err := jobs.Get(c, jobId)
 		if err != nil {
 			if _, ok := err.(golangsdk.ErrDefault404); ok {
```
Comment (hmbig2) [inline huaweicloud/resource_huaweicloud_compute_volume_attach.go:187]: when creating , this error will cause an infinite loop

File: huaweicloud/resource_huaweicloud_compute_volume_attach.go [hunk 4]
```diff
@@ -104,20 +107,17 @@ func resourceComputeVolumeAttachV2Create(d *schema.ResourceData, meta interface{
...
 
-	return resourceComputeVolumeAttachV2Read(d, meta)
+	return resourceComputeVolumeAttachRead(d, meta)
 }
 
-func resourceComputeVolumeAttachV2Read(d *schema.ResourceData, meta interface{}) error {
+func resourceComputeVolumeAttachRead(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*config.Config)
-	computeClient, err := config.ComputeV2Client(GetRegion(d, config))
+	computeClient, err := config.ComputeV1Client(GetRegion(d, config))
```
Comment (hmbig2) [inline huaweicloud/resource_huaweicloud_compute_volume_attach.go]: declare a region may be better , can reused in line 141

File: huaweicloud/resource_huaweicloud_compute_volume_attach.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/resource_huaweicloud_compute_volume_attach.go:9]: all volumeattach packages have been removed.

File: huaweicloud/resource_huaweicloud_compute_volume_attach.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/resource_huaweicloud_compute_volume_attach.go]: done

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccComputeVolumeAttach'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccComputeVolumeAttach -timeout 360m -parallel 4
=== RUN   TestAccComputeVolumeAttach_basic
=== PAUSE TestAccComputeVolumeAttach_basic
=== RUN   TestAccComputeVolumeAttach_device
=== PAUSE TestAccComputeVolumeAttach_device
=== CONT  TestAccComputeVolumeAttach_basic
=== CONT  TestAccComputeVolumeAttach_device
--- PASS: TestAccComputeVolumeAttach_device (253.61s)
--- PASS: TestAccComputeVolumeAttach_basic (259.14s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       259.367s
```
```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccComputeVolumeAttach'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccComputeVolumeAttach -timeout 360m -parallel 4
=== RUN   TestAccComputeVolumeAttach_basic
=== PAUSE TestAccComputeVolumeAttach_basic
=== RUN   TestAccComputeVolumeAttach_device
=== PAUSE TestAccComputeVolumeAttach_device
=== CONT  TestAccComputeVolumeAttach_basic
=== CONT  TestAccComputeVolumeAttach_device
--- PASS: TestAccComputeVolumeAttach_device (241.03s)
--- PASS: TestAccComputeVolumeAttach_basic (254.05s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       254.582s
```

File: huaweicloud/resource_huaweicloud_compute_volume_attach.go [hunk 5]
```diff
@@ -127,36 +128,27 @@ func resourceComputeVolumeAttachV2Read(d *schema.ResourceData, meta interface{})
 		return err
 	}
 
-	attachment, err := volumeattach.Get(computeClient, instanceId, attachmentId).Extract()
+	attachment, err := block_devices.Get(computeClient, instanceId, attachmentId).Extract()
```
Comment (Jason-Zhang9309) [inline huaweicloud/resource_huaweicloud_compute_volume_attach.go]: suggest to replace attachmentId volumeId
also do it in functions parseComputeVolumeAttachmentId

File: huaweicloud/services/evs/resource_huaweicloud_evs_volume.go [hunk 6]
```diff
@@ -431,24 +432,24 @@ func resourceEvsVolumeDelete(ctx context.Context, d *schema.ResourceData, meta i
 		}
 		for _, attachment := range v.Attachments {
 			logp.Printf("[DEBUG] The attachment is: %v", attachment)
-			err = volumeattach.Delete(computeClient, attachment.ServerID, attachment.AttachmentID).ExtractErr()
+			opts := block_devices.DetachOpts{
+				ServerId: attachment.ServerID,
+			}
+			job, err := block_devices.Detach(computeClient, attachment.AttachmentID, opts)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/evs/resource_huaweicloud_evs_volume.go:438]: the client version is not correct

File: huaweicloud/services/evs/resource_huaweicloud_evs_volume.go [hunk 7]
```diff
@@ -431,24 +432,24 @@ func resourceEvsVolumeDelete(ctx context.Context, d *schema.ResourceData, meta i
...
 			if err != nil {
-				return fmtp.DiagErrorf("An error occurred while detaching the volume from the object instance: %s", err)
+				return diag.FromErr(err)
+			}
+			stateConf := &resource.StateChangeConf{
+				Pending:    []string{"RUNNING"},
+				Target:     []string{"SUCCESS", "NOTFOUND"},
+				Refresh:    AttachmentJobRefreshFunc(computeClient, job.ID),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/evs/resource_huaweicloud_evs_volume.go:445]: the client version is not correct

File: huaweicloud/provider.go [hunk 8]
```diff
@@ -473,7 +473,7 @@ func Provider() *schema.Provider {
 			"huaweicloud_compute_keypair":                  ResourceComputeKeypairV2(),
 			"huaweicloud_compute_servergroup":              ResourceComputeServerGroupV2(),
 			"huaweicloud_compute_eip_associate":            ResourceComputeFloatingIPAssociateV2(),
-			"huaweicloud_compute_volume_attach":            ResourceComputeVolumeAttachV2(),
+			"huaweicloud_compute_volume_attach":            ResourceComputeVolumeAttach(),
```
Comment (niuzhenguo) [inline huaweicloud/provider.go:477]: suggest to keep the name to `ResourceComputeVolumeAttachV2`, this change will break other clouds.

Comment (Lance52259) [reply huaweicloud/provider.go:477]: The v2 version interface is not used under this resource now, so the v2 suffix is inaccurate.

File: huaweicloud/services/evs/resource_huaweicloud_evs_volume.go (hunk 7 above)
Comment (Lance52259) [reply huaweicloud/services/evs/resource_huaweicloud_evs_volume.go:445]: fixed

File: huaweicloud/services/evs/resource_huaweicloud_evs_volume.go (hunk 6 above)
Comment (Lance52259) [reply huaweicloud/services/evs/resource_huaweicloud_evs_volume.go:438]: fixed

File: huaweicloud/resource_huaweicloud_compute_volume_attach.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/resource_huaweicloud_compute_volume_attach.go]: yeah, the `attachmentId` is inaccurate.

**Comment by Lance52259** [comment]
> Can we update the resource with context and diag by the way?

This change will be revised through other PR in the future.

**Comment by Lance52259** [comment]
Dependency package `compute/v2/extensions/volumeattach` has been removed from the `/vendor` dir.

**Pull request merged**